x/participationrewards/keeper/rewards_validatorSelection.go

```diff
@@ -240,3 +268,56 @@ func (k Keeper) calcOverallScores(
 
 	return nil
 }
+
+func (k Keeper) calcUserAllocations(
+	ctx sdk.Context,
+	zone icstypes.RegisteredZone,
+	zs zoneScore,
+	za map[string]sdk.Coins,
+) ([]userAllocation, error) {
+	k.Logger(ctx).Info("calcUserAllocations", "zone", zone.ChainId, "scores", zs, "allocations", za)
+
+	userAllocations := make([]userAllocation, 0)
+
+	type userScore struct {
+		address string
+		score   sdk.Dec
+	}
+
+	sum := sdk.NewDec(0)
+	userScores := make([]userScore, 0)
+	for _, di := range k.icsKeeper.AllOrdinalizedIntents(ctx, zone) {
```

Comment:
This is fine for now, but will need to be replaced with the snapshotted intents from the previous boundary.